### ISSUE TITLE
deno lint doesn't detect that I forgot to import "foo" before calling foo()

### ISSUE THREAD

**Issue by mo** [denoland/deno_lint#1342]
![image](https://github.com/user-attachments/assets/2690952f-976b-473d-8198-8d74b3af9c16)

echo -e "foo()\nlet x" > test.js
deno lint test.js

Deno lint finds the unused "x" but it doesn't warn me that I forgot to import foo().

If I try eslint (npx eslint test.js) instead, it immediately says:
1:1  error  'foo' is not defined           no-undef

i'm using deno 2.0.6

**Comment by mo** [comment]
I also did "deno lint --rules" and "no-undef" was NOT included in the list on available rules. However, when I added it to my deno.json it started to work properly. How are newcomers to deno project supposed to understand this?

```
  "lint": {
    "rules": {
      "include": [
        "no-undef"
      ]
    }
  },
```

**Comment by mo** [comment]
hmm, I found this comment now:
https://github.com/denoland/deno_lint/issues/630#issuecomment-810286231

...what were the arguments against no-undef in recommended?
